docs/source/_src_user_guide/state_propagation/propagation_setup/propagation_architecture.rst

```diff
@@ -0,0 +1,81 @@
+* If a termination condition was reached *during* one of the sub-stages of the time step, the propagation is stopped, and the results returned. Note that this only happens if the ``assess_termination_on_minor_steps`` input to one of the integrator setting functions in :mod:`~tudatpy.numerical_simulation.propagation_setup.integrator` is set to true (false by default)    
+* If output is to be saved at the current time step (default: saved every time step):
+
+  * The pair :math:`(t_{i+1},\mathbf{x}_{i+1})` is added to the propagated state history
+  * If any dependent variables are to be saved, the environment is updated to the current time/state :math:`(t_{i+1},\mathbf{x}_{i+1})`, see :ref:`single_propagation_evaluation`, and the dependent variables are extracted.
+* It is checked whether the :math:`(t_{i+1},\mathbf{x}_{i+1})` pair meets the termination conditions. If the termination conditions are exceeded, and the ``terminate_exactly_on_final_condition`` input to the termination condition settings is set to false (see :mod:`~tudatpy.numerical_simulation.propagation_setup.propagator`), the propagation is finished, and the results are returned. If this variable is set to true:
+
+  * If the termination condition is a given time (:func:`~tudatpy.numerical_simulation.propagation_setup.propagator.time_termination`), the final time step is adjusted such that the final time is reached exactly
+  * If the termination confition is a given dependent variable value (:func:`~tudatpy.numerical_simulation.propagation_setup.propagator.dependent_variable_termination`), a root finding algorithm is used to iterate to the time :math:`t_{i+1}` at which the given value is achieved.
+* In either case, the :class:`~tudatpy.numerical_simulation.propagation.PropagationTerminationReason` is set to `termination_condition_reached``, and the state and dependent variable history is returned.
+
+.. _propagator_post_processing:
+
+Propagator post-processing
+==========================
+
+After the propagation is finished, the following post-processing steps are performed before returning the simulation to the user:
+
+* The propagated states are converted to conventional states. After the propagation, the time histories of both may be extracted from the :attr:`~tudatpy.numerical_simulation.SingleArcSimulator.unprocessed_state_history` and :attr:`~tudatpy.numerical_simulation.SingleArcSimulator.state_history` atributes, respectively
+* If the ``set_integrated_result`` to the :class:`~tudatpy.numerical_simulation.SingleArcSimulator` is set to true, the propagated states (in conventional formulation) are used to reset the environment of the propagated body/bodies. For the different state types, this means:
```

Comment:
Same comment as above for the set_integrated_result functionality: might be worth explicitely stating that the ephemeris is updated to the propagated solution (for new tudat users)?